### PR TITLE
redolog: close tmp file after use & refactor

### PR DESCRIPTION
**Pull request by ben1009**
### What problem does this PR solve? 
close https://github.com/pingcap/ticdc/issues/3303

### What is changed and how it works?
- close tmp file after sync when write meta file
- modify the shouOpen logic then filter out event later in reader when apply logs

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
 None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
/cc 

I will take a look tonight.

File: cdc/redo/reader/file.go [hunk 1]
```diff
@@ -361,7 +361,9 @@ func shouldOpen(startTs, endTs uint64, name, fixedType string) (bool, error) {
 	if filepath.Ext(name) == common.TmpEXT {
 		return true, nil
 	}
-	return commitTs > startTs && commitTs <= endTs, nil
+	// the commitTs=max(ts of log item in the file), if max > startTs then should open,
+	// filter out ts in (startTs, endTs] for consume
+	return commitTs > startTs, nil
```
Comment (amyangfei) [inline cdc/redo/reader/file.go:366]: Does this mean we could loss data when applying redo logs before this PR?

Comment (ben1009) [reply cdc/redo/reader/file.go:366]: yes

File: cdc/redo/reader/file_test.go [hunk 2]
```diff
@@ -183,8 +185,7 @@ func TestReaderOpenSelectedFiles(t *testing.T) {
 			args: arg{
 				dir:       dir,
 				fixedName: common.DefaultDDLLogFileType,
-				startTs:   0,
-				endTs:     9,
```
Comment (hi-rustin) [inline cdc/redo/reader/file_test.go:187]: I think we can remove this parameter now.

File: cdc/redo/reader/reader.go [hunk 3]
```diff
@@ -231,7 +231,8 @@ func (l *LogReader) ReadNextLog(ctx context.Context, maxNumberOfEvents uint64) (
 	for l.rowHeap.Len() != 0 && i < maxNumberOfEvents {
 		item := heap.Pop(&l.rowHeap).(*logWithIdx)
 		if item.data.RedoRow != nil && item.data.RedoRow.Row != nil &&
-			item.data.RedoRow.Row.CommitTs > l.cfg.startTs {
+			item.data.RedoRow.Row.CommitTs > l.cfg.startTs &&
```
Comment (hi-rustin) [inline cdc/redo/reader/reader.go:234]: Perhaps a comment is needed as to exactly what situation it deals with.

File: cdc/redo/reader/reader.go [hunk 4]
```diff
@@ -292,7 +293,8 @@ func (l *LogReader) ReadNextDDL(ctx context.Context, maxNumberOfEvents uint64) (
 	for l.ddlHeap.Len() != 0 && i < maxNumberOfEvents {
 		item := heap.Pop(&l.ddlHeap).(*logWithIdx)
 		if item.data.RedoDDL != nil && item.data.RedoDDL.DDL != nil &&
-			item.data.RedoDDL.DDL.CommitTs > l.cfg.startTs {
+			item.data.RedoDDL.DDL.CommitTs > l.cfg.startTs &&
```
Comment (hi-rustin) [inline cdc/redo/reader/reader.go:296]: Perhaps a comment is needed as to exactly what situation it deals with. ditto

**Review by hi-rustin — COMMENTED** [review]
reset LGTM

**Review by amyangfei — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**